docker-compose.yaml

```diff
@@ -1,11 +1,11 @@
 version: '3.8'
 services:
-  mysql:
+  mongo:
     image: 'mongo:latest'
     ports:
       - "10640:27017"
   redis-cluster:
-    image: 'grokzen/redis-cluster:latest'
+    image: 'grokzen/redis-cluster:7.0.7'
```

Comment:
Redis 최신 버전을 사용하면 어플리케이션에서 레디스에 접근 시 다음과 같은 예외가 발생한다.
```
[redis://redis-:7777]: Unable to connect to [redis-/:7777]: DENIED Redis is running in protected mode because protected mode is enabled and no password is set for the default user. In this mode connections are only accepted from the loopback interface. If you want to connect from external computers to Redis you may adopt one of the following solutions: 1) Just disable protected mode sending the command 'CONFIG SET protected-mode no' from the loopback interface by connecting to Redis from the same host the server is running, however MAKE SURE Redis is not publicly accessible from internet if you do so. Use CONFIG REWRITE to make this change permanent. 2) Alternatively you can just disable the protected mode by editing the Redis configuration file, and setting the protected mode option to 'no', and then restarting the server. 3) If you started the server manually just for testing, restart it with the '--protected-mode no' option. 4) Setup a an authentication password for the default user. NOTE: You only need to do one of the above things in order for the server to start accepting connections from the outside.
```

해당 예외는 Redis 기본 유저에 비밀번호를 설정하지 않아서 발생하는 문제이다.

레디스가 접근할 수 있도록 하려면 protected-mode를 no로 설정해야 한다.
latest 버전에서는 protected-mode를 docker-compose로 해결할 수 없다.
Docker Hub의 Image Description에서는 latest == 7.0.7이라고 나와있지만,  latest가 7.0.7이 아닌 듯하다.

7.0.7 버전으로 직접 명시 후 이미지를 다운받아 사용하면, protected-mode no 로 설정된 레디스를 사용할 수 있다.